README.md
PastaCRM


CHANGELOG
██████╗  █████╗ ███████╗████████╗ █████╗ ███╗   ███╗███████╗██████╗ ██╗ █████╗      ██████╗███╗   ███╗███████╗
██╔══██╗██╔══██╗██╔════╝╚══██╔══╝██╔══██╗████╗ ████║██╔════╝██╔══██╗██║██╔══██╗    ██╔════╝████╗ ████║██╔════╝
██████╔╝███████║███████╗   ██║   ███████║██╔████╔██║█████╗  ██║  ██║██║███████║    ██║     ██╔████╔██║███████╗
██╔═══╝ ██╔══██║╚════██║   ██║   ██╔══██║██║╚██╔╝██║██╔══╝  ██║  ██║██║██╔══██║    ██║     ██║╚██╔╝██║╚════██║
██║     ██║  ██║███████║   ██║   ██║  ██║██║ ╚═╝ ██║███████╗██████╔╝██║██║  ██║    ╚██████╗██║ ╚═╝ ██║███████║
╚═╝     ╚═╝  ╚═╝╚══════╝   ╚═╝   ╚═╝  ╚═╝╚═╝     ╚═╝╚══════╝╚═════╝ ╚═╝╚═╝  ╚═╝     ╚═════╝╚═╝     ╚═╝╚══════╝

   _____ _                              _             
  / ____| |                            | |            
 | |    | |__   __ _ _ __   __ _  ___  | | ___   __ _ 
 | |    | '_ \ / _` | '_ \ / _` |/ _ \ | |/ _ \ / _` |
 | |____| | | | (_| | | | | (_| |  __/ | | (_) | (_| |
  \_____|_| |_|\__,_|_| |_|\__, |\___| |_|\___/ \__, |
                            __/ |                __/ |
                           |___/                |___/ 

#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#
===================
======
==  ver 2.00
== 2018-03-15 -> 2018-03-18
======
===================
=====

Done Things:
- Calkowite przebudowanie back-end, struktury bazy danych, ujednolicenie UX.
- Usuniecie calosci modulu zadan ToDo.

#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#
===================
======
==  ver 1.18
== 2018-01-15
======
===================
=====

Done Things:
- Poprawki cennika i wyswietlania cen
- Poprawki UI oraz UX

Updated Files:
├── functions.php
├── dashboard.php
├── pricingList.php
├── applications.php
├── websitesMore.php
└── changelog.txt

#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#
===================
======
==  ver 1.17
== xxxx-xx-xx
======
===================
=====

Done Things:
- Usunięcie Freedcamp z dashboard, dodanie WakaTime.
- Dodanie statystyk programowania WakaTime do dashboard.
- Dodanie submenu "Applications".
- Spolszczenie CMS.
- Usuniecie "Templates Wordpress" i "Templates Static" - zamiast nich "Szablony dokumentów".
- Reorganizacja menu.
- Standaryzacja funkcji, małe poprawki jakości kodu.

#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#
===================
======
==  ver 1.16
== 2017-12-27
======
===================
=====

Done Things:
- Standaryzacja UI
- Nowe narzedze w Useful Tools: WinBase
- Przeniesienie logo z UsefulTools do folderu img/UsefulTools/
- Wyrzucenie rozwijanej listy w headerze, zamiast tego jedna pozycja: logout.
- Inne poprawki UX i UI.
- Fix kolorystyki wykresu z podsumowaniem zarobkow w kolejnych latach.
- Fix wysyłania plików - zamiana '|' na '-'.
- Normalizacja nazewnictwa funkcji.
- Rozbudowanie danych zapisywanych do logu.

#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#
===================
======
==  ver 1.15
== 2017-12-07
======
===================
=====

Done Things:
- Standaryzacja UI

#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#
===================
======
==  ver 1.14
== 2017-12-06
======
===================
=====

Done Things:
- Zmiana tabeli ze statystykami płatności na dwa wykresy:
-- a) Wykres płatności za strony / miesiąc
-- b) Wykres płatności za serwis / miesiąc
- Jesli header nie przekierowuje, to przekieruje meta refresh (po zapisaniu)
- Dodanie przycisku "Generuj podsumowanie pracy" / "Generate work summary"
    (na razie disabled) do computerServiceEdit
- Modyfikacja UI help, podział na kategorie, dodanie nowych poradnikow
- Wykres typu 'pie' z zestawieniem:
-- a) Zarobki ze stron
-- b) Zarobki z serwisu komputerowego
- Wykres z historia zarobkow w kolejnych latach
- Zamiast tabeli z wplatami w dashboard dodany wykres na 6 miesiecy
- W gornym pasku usuniety link Games + porzadek w kodzie
- Porzadek w kodzie ComputerServiceMore

#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#
===================
======
==  ver 1.13
== 2017-11-25
======
===================
=====

Done Things:
- Added 'Delete server' function

#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#
===================
======
==  ver 1.12
== 2017-11-14
======
===================
=====

Done Things:
- Added "UptimeRobot" button to dashboard
- Deleted value from Computer Service services list
- Payment charts: rounded values, because I saw 499.9900016784668 PLN xD

#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#
===================
======
==  ver 1.11
== 2017-11-04
======
===================
=====

Done Things:
- Add https://uncss-online.com/ to UsefullTools
- Repaired payments chart and stats [part]
- Added support for float payments, eg. 49.99

#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#
===================
======
==  ver 1.10
== 2017-10-15
======
===================
=====

Done Things:
- Dodanie znacznika [br] do BB-Code
- Screeny w templates other - kod stron HTML: Coming Soon + Block
- Średnie zarobki w transactionsMore
- WebsitesCalendar ID > Unique ID + standaryzacja funkcji.
- Fix wyswietlania logu strony - dodanie BB-Code
- Usuniece czerwonego tla z dropped websites.
- Autofocus na login
- Strona under construction powinna byc wyraznie
 oznaczona w podgladzie serwerow. Podglad serwerow nie pokazuje
 rowniez stron ze statusem "dropped".
- Walidacja HTML na websites.php. Fix poprawnosci kodu.
- Jesli Computer Service ma status Under Contruction,
 system ma jej NIE wliczac do zestawienia platnosci.
- Pricing List Basic sprawdza Android ID. Jesli nie jest to ID
 Xiaomi MI4 to nie wyswietla danych.
- Fix dashboard: Nowo dodane zadanie pokazuje sie od razu, a nie 
 dopiero po odswiezeniu strony.
- Usuniecie liczby "less than 46 days" z komunikatu o wygasaniu
 serwera. Zastapienie jej przez "less than one and half month to"

======
===================
#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#=#
===================
======
==  ver 1.9
== 2017-10-07
======
===================
=====

Done Things:
- Site Log: Poszerzenie kolumny "type" & "date" >> "Time".
- Nowy design panelu logowania: Material Design.
- Zwiększenie ilości znaków w opisie stony (baza danych)

poll.php
<?php
@session_start();
require_once("inc/functions.php");
if(checkUserPermissions(3)){
    $siteTitle = "Ankieta";
?>
<!DOCTYPE html>
<html>
<head>
    <title>
        <?= showSiteTitle($siteTitle) ?>
    </title>
    <?php include_once("inc/head.php"); ?>
</head>

<body>
    <div id="wrapper">
        <?php include_once("inc/menus.php"); ?>
        <div id="page-wrapper">
            <div class="container-fluid">

            <!-- HEADER -->
            <div class="row">
                <div class="col-lg-12">
                    <h1 class="page-header">
                        <?= $siteTitle ?>
                        <small>Jak klienci oceniają wykonaną pracę?</small>
                    </h1>
                    <ol class="breadcrumb">
                        <li>
                            <i class="fa fa-dashboard"></i>
                            <a href="dashboard.php">Dashboard</a>
                        </li>
                        <li class="active">
                            <i class="fa fa-pie-chart"></i>
                            <?= $siteTitle ?>
                        </li>
                    </ol>
                </div>
            </div>

<!-- STRONA -->
<div class="row">
    <div class="col-md-8">

        <!-- PANEL: STATYSTYKI OGOLNE -->
        <div class="panel panel-default">
            <div class="panel-heading">
                <i class="fa fa-line-chart"></i> Garść ogólnych statystyk
            </div>
            <div class="panel-body">

                <div class="alert alert-warning">
                    <b>Panel w trakcie budowy...</b>
                </div>

                <!-- Wykres 1 --
                <div class="col-lg-6">
                    <div class="panel panel-default">
                        <div class="panel-body">
                            <?php
                            /*
                                $obj_data = getPollResults("all_satisfaction");

                                $data = array("60", "4");
                                $title = "Czy klienci są zadowoleni z usługi?";
                                $labels = array("Tak", "Nie");
                                $colors = array("#4CAF50", "#F44336");
                                
                                showChart("doughnut", $title, $data, $labels, $colors);
                                */
                            ?>
                        </div>
                    </div>
                </div>

                <!-- Wykres 2 --
                <div class="col-lg-6">
                    <div class="panel panel-default">
                        <div class="panel-body">
                            <?php
                            /*
                                $title = "Ilość głosów w kolejnych miesiącach";
                                $dane = array(5, 10, 15, 20, 25, 40, 50, 68, 80, 90, 95, 105);
                                $labels = array("01", "02", "03", "04", "05", "06", "07", "08", "09", "10", "11", "12");
                                
                                showChart("lineSoft", $title, $dane, $labels);
                                */
                            ?>
                        </div>
                    </div>
                </div>

            </div>
        </div>

        <!-- PANEL STRONY --
        <div class="panel panel-default">
            <div class="panel-heading">
                <i class="fa fa-globe"></i> Statystyki dla stron internetowych
            </div>
            <div class="panel-body">

                <!-- Wykres 1 --
                <div class="col-lg-6">
                    <div class="panel panel-default">
                        <div class="panel-body">
                            <?php
                            /*
                                $title = "Czy klienci są zadowoleni z usługi? (strony internetowe)";
                                //$dane = getPollResults("websites");
                                $dane = array("60", "10");
                                $labels = array("Tak", "Nie");
                                $colors = array("#4CAF50", "#F44336");
                                
                                showChart("doughnut", $title, $dane, $labels, $colors);
                              */  
                            ?>
                        </div>
                    </div>
                </div>

                <!-- Wykres 2 --
                <div class="col-lg-6">
                    <div class="panel panel-default">
                        <div class="panel-body">
                            <?php
                            /*
                                $title = "Łączna ilość głosów w kolejnych miesiącach";
                                //$dane = getPollResults("websites");
                                $dane = array(5, 10, 15, 20, 25, 40);
                                $labels = array("Styczeń", "Luty", "Marzec", "Kwiecień", "Maj", "Czerwiec");
                                
                                showChart("line", $title, $dane, $labels);
                                */
                            ?>
                        </div>
                    </div>
                </div>

                <!-- Wykres 3 --
                <div class="col-lg-6">
                    <div class="panel panel-default">
                        <div class="panel-body">
                            <?php
                            /*
                                $title = "Ocena czasu wykonania usługi";
                                //$dane = getPollResults("websites");
                                $dane = array("60",
                                                "50",
                                                "40",
                                                "30",
                                                "20",
                                                "10");
                                $labels = array("6 - Perfekcyjnie",
                                                "5 - Bardzo dobrze",
                                                "4 - Dobrze",
                                                "3 - Średnio",
                                                "2 - Źle",
                                                "1 - Bardzo źle");
                                $colors = array("#4CAF50",
                                                "#81C784",
                                                "#A5D6A7",
                                                "#FFEB3B",
                                                "#EF5350",
                                                "#F44336");
                                showChart("doughnut", $title, $dane, $labels, $colors);
                                */
                            ?>
                        </div>
                    </div>
                </div>

                <!-- Wykres 4 --
                <div class="col-lg-6">
                    <div class="panel panel-default">
                        <div class="panel-body">
                            <?php
                            /*
                                $title = "Ocena jakości wykonanej usługi";
                                //$dane = getPollResults("websites");
                                $dane = array("60",
                                                "50",
                                                "40",
                                                "30",
                                                "20",
                                                "10");
                                $labels = array("6 - Perfekcyjnie",
                                                "5 - Bardzo dobrze",
                                                "4 - Dobrze",
                                                "3 - Średnio",
                                                "2 - Źle",
                                                "1 - Bardzo źle");
                                $colors = array("#4CAF50",
                                                "#81C784",
                                                "#A5D6A7",
                                                "#FFEB3B",
                                                "#EF5350",
                                                "#F44336");
                                showChart("doughnut", $title, $dane, $labels, $colors);
                                */
                            ?>
                        </div>
                    </div>
                </div>

                <!-- Wykres 5 --
                <div class="col-lg-6">
                    <div class="panel panel-default">
                        <div class="panel-body">
                            <?php
                            /*
                                $title = "Ocena komunikacji podczas zlecenia";
                                //$dane = getPollResults("websites");
                                $dane = array("60",
                                                "50",
                                                "40",
                                                "30",
                                                "20",
                                                "10");
                                $labels = array("6 - Perfekcyjnie",
                                                "5 - Bardzo dobrze",
                                                "4 - Dobrze",
                                                "3 - Średnio",
                                                "2 - Źle",
                                                "1 - Bardzo źle");
                                $colors = array("#4CAF50",
                                                "#81C784",
                                                "#A5D6A7",
                                                "#FFEB3B",
                                                "#EF5350",
                                                "#F44336");
                                showChart("doughnut", $title, $dane, $labels, $colors);
                                */
                            ?>
                        </div>
                    </div>
                </div>

                <!-- Wykres 6 --
                <div class="col-lg-6">
                    <div class="panel panel-default">
                        <div class="panel-body">
                            <?php
                                /*
                                $title = "Ocena ceny za wykonaną usługę";
                                //$dane = getPollResults("websites");
                                $dane = array("60",
                                                "50",
                                                "40",
                                                "30",
                                                "20",
                                                "10");
                                $labels = array("6 - Perfekcyjnie",
                                                "5 - Bardzo dobrze",
                                                "4 - Dobrze",
                                                "3 - Średnio",
                                                "2 - Źle",
                                                "1 - Bardzo źle");
                                $colors = array("#4CAF50",
                                                "#81C784",
                                                "#A5D6A7",
                                                "#FFEB3B",
                                                "#EF5350",
                                                "#F44336");
                                showChart("doughnut", $title, $dane, $labels, $colors);
                                */
                            ?>
                        </div>
                    </div>
                </div>
-->
            </div>
        </div>

    </div>
    <div class="col-md-4">
        <div class="panel panel-default">
            <div class="panel-heading">
                <i class="fa fa-life-ring"></i> Pomoc</div>
                <div class="panel-body">
                    <div class="alert alert-info">
                        <s>Statystyki są pobierane z zewnętrznego serwera i odświeżane tylko raz w ciągu każdego dnia</s>
                    </div>
                    <hr>
                    <button class="btn btn-warning full-width" disabled>Wymuś pobranie nowych danych z serwera</button>
                    <hr>
                    <button class="btn btn-primary full-width" disabled>Pobierz statystyki jako plik PDF</button>
                    <hr>
                    <button class="btn btn-primary full-width" disabled>Pobierz statystyki jako RAW</button>
                    <hr>
                    <button class="btn btn-primary full-width" onclick="javascript:window.print();">Drukuj stronę ze statystykami</button>
                </div>
            </div>
        </div>
    </div>
</div>
<!-- KONIEC STRONY -->

</div>
</div>
</body>
</html>
<?php } ?>